### PR TITLE
Tensor core

### PR DESCRIPTION
**Pull request by gaoteng-git**
Add "TC Eligible", "TC Self(%)" and "TC Total(%)" to Operator view.
image
image

Add "TC Used" to Kernel view grouped by "Kernel Name".
image

Add "Kernel Uses TC" and "Op is TC Eligible" to Kernel view grouped by "Kernel Properties + Op Name".
image

### CODE REVIEW

File: tb_plugin/torch_tb_profiler/profiler/kernel_parser.py [hunk 1]
```diff
@@ -5,14 +5,20 @@
...
+from .tensor_core import TC_Whitelist
 
 
 class KernelParser:
     def __init__(self):
         self.kernel_stat = None
 
     def parse_events(self, events):
-        events = [vars(event) for event in events if event.type == EventTypes.KERNEL ]
+        events_dict = []
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/kernel_parser.py:16]: add to event?

File: tb_plugin/torch_tb_profiler/profiler/node.py [hunk 2]
```diff
@@ -160,6 +180,7 @@ def __init__(self, name, start_time, end_time, type, tid, external_id=None,
         self.block = block
         self.regs_per_thread = regs_per_thread
         self.shared_memory = shared_memory
+        self.tc_used = self.name in TC_Whitelist() if self.type == EventTypes.KERNEL else False
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/node.py]: can we reuse the tc_used set in event?

Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/node.py]: is it possible that not create the object?

File: tb_plugin/torch_tb_profiler/profiler/node.py [hunk 3]
```diff
@@ -72,6 +76,9 @@ def __init__(self, name, start_time, end_time, type, tid, external_id=None, devi
         self.self_device_duration = self_device_duration
         self.memory_records = []
         # self.parent_node = None
+        self.tc_eligible = False
+        self.tc_self_duration = 0
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/node.py:80]: add comments about the meaning of each field

File: tb_plugin/torch_tb_profiler/profiler/node.py [hunk 4]
```diff
@@ -100,15 +107,27 @@ def fill_stats(self):
             rt.update_device_op_node(self)
 
         self.self_host_duration = self.end_time - self.start_time
+        self.tc_eligible = TC_OP_Whitelist.is_fit(self.name)
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/node.py:110]: move to init?

File: tb_plugin/torch_tb_profiler/profiler/tensor_core.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+class TC_Whitelist:
+    # Refer to https://github.com/NVIDIA/PyProf/blob/fd1b2902e3306119eee40ba6b6e8b2f816920c29/pyprof/prof/tc.py#L19
+    whitelist = ['h884', 's884', 'h1688', 's1688', 'hmma', 'i8816', '16816',
+                 'dgrad_1x1_stride_2x2', 'first_layer_wgrad_kernel', 'conv1x1',
+                 'conv2d_c1_k1', 'direct_group', 'xmma_implicit_gemm',
+                 'xmma_sparse_conv', 'xmma_warp_specialized_implicit_gemm',
+                 'xmma_gemm', 'xmma_sparse_gemm', 'c1688']
+
+    @classmethod
+    def is_fit(cls, item):
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/tensor_core.py:13]: refer to https://github.com/python/cpython/blob/main/Lib/enum.py#L609

**Comment by nbcsm** [comment]
please make sure doc and tooltips are updated as well.

File: tb_plugin/torch_tb_profiler/profiler/kernel_parser.py (hunk 1 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/kernel_parser.py]: Thanks. Done

File: tb_plugin/torch_tb_profiler/profiler/node.py (hunk 2 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/node.py]: Thanks. Done.

File: tb_plugin/torch_tb_profiler/profiler/node.py (hunk 3 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/node.py]: Thanks. Done.

File: tb_plugin/torch_tb_profiler/profiler/node.py (hunk 4 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/node.py]: Thanks. Done.

File: tb_plugin/torch_tb_profiler/profiler/tensor_core.py (hunk 5 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/tensor_core.py]: Thanks. Done.

**Comment by gaoteng-git** [comment]
> please make sure doc and tooltips are updated as well.

Got it. I prefer to do those in another PR.

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py [hunk 6]
```diff
@@ -153,6 +160,9 @@ def aggregate(key_to_agg, key, op):
                 agg.device_duration += op.device_duration
                 agg.self_host_duration += op.self_host_duration
                 agg.self_device_duration += op.self_device_duration
+                agg.tc_eligible = op.tc_eligible
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/module_parser.py:163]: can be moved into constructor?

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py [hunk 7]
```diff
@@ -46,9 +50,12 @@ def __init__(self, kernel, op_name):
         self.total_duration = 0
         self.min_duration = sys.maxsize
         self.max_duration = 0
-
         self.blocks_per_sm = 0.0
         self.occupancy = 0.0
+        self.tc_used = False
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/module_parser.py:55]: needless?

File: tb_plugin/torch_tb_profiler/profiler/run_generator.py [hunk 8]
```diff
@@ -314,7 +317,9 @@ def _generate_kernel_op_table(self):
                             {"type": "string", "name": "Grid"},
                             {"type": "string", "name": "Block"},
                             {"type": "number", "name": "Register Per Thread"},
-                            {"type": "number", "name": "Shared Memory"}]
+                            {"type": "number", "name": "Shared Memory"},
+                            {"type": "string", "name": "Kernel Uses TC"},
+                            {"type": "string", "name": "Op is TC Eligible"}]
```
Comment (guotuofeng) [inline tb_plugin/torch_tb_profiler/profiler/run_generator.py:322]: better use full word for tooltip

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py (hunk 6 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/module_parser.py]: Thanks. Done.

File: tb_plugin/torch_tb_profiler/profiler/module_parser.py (hunk 7 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/module_parser.py]: Thanks. Done.

File: tb_plugin/torch_tb_profiler/profiler/run_generator.py (hunk 8 above)
Comment (gaoteng-git) [reply tb_plugin/torch_tb_profiler/profiler/run_generator.py]: Thanks. Done.

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**